R/provider-ollama.R

```diff
@@ -47,25 +47,41 @@ chat_ollama <- function(system_prompt = NULL,
       i = "Locally installed models: {.str {models}}."
     ))
   }
+
+  turns <- normalize_turns(turns, system_prompt)
   echo <- check_echo(echo)
 
-  chat_openai(
-    system_prompt = system_prompt,
-    turns = turns,
+  provider <- ProviderOllama(
     base_url = file.path(base_url, "v1"), ## the v1 portion of the path is added for openAI compatible API
-    api_key = "ollama", # ignored
     model = model,
     seed = seed,
-    api_args = api_args,
-    echo = echo
+    extra_args = api_args,
+    api_key = "ollama" # ignored
   )
+
+  Chat$new(provider = provider, turns = turns, echo = echo)
 }
 
-chat_ollama_test <- function(..., model = "llama3.3") {
+ProviderOllama <- new_class(
+  "ProviderOllama",
+  parent = ProviderOpenAI,
+  properties = list(
+    api_key = prop_string(),
+    model = prop_string(),
+    seed = prop_number_whole(allow_null = TRUE)
+  )
+)
+
+chat_ollama_test <- function(..., model = "llama3.2:1b") {
   if (!has_ollama()) {
     testthat::skip("ollama not found")
   }
 
+  testthat::skip_if_not(
```

Comment:
Would you mind adding a test, even if it's not going to run on CI?

Reply:
Tests added in d3b42a12aaa408090027cb494e5ab4b47e44ac5f. I added one test around the shape of the tool def and one smoke test that `chat_ollama()` still works when a tool with optional arguments is registered.